README.md

```diff
@@ -0,0 +1,130 @@
+
+## Jenkins Job Builder
+
+[Jenkins Job Builder](https://docs.openstack.org/infra/jenkins-job-builder/) (JJB) is an OpenStack tool for generating Jenkins job definitions (an XML file) from a set of YAML job descriptions, which we store in version control.
+
+* [JJB YAML files in foreman-infra](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files)
+
+Puppet deploys these onto our Jenkins server (a recursive file copy) and when they change, it runs the JJB tool to update the jobs in the live instance.  It also refreshes them daily to overwrite manual changes.
+
+Useful resources:
+
+* [Job definitions, templates etc.](https://docs.openstack.org/infra/jenkins-job-builder/definition.html)
+* [Modules, e.g. SCM, publishers, builders](https://docs.openstack.org/infra/jenkins-job-builder/definition.html#modules)
+
+## Jenkins Job Naming conventions
+
+**Note** Because `centos.org` is a shared environment all jobs are prefixed by `foreman-` to denote they're ours.
+
+| **Name**                | **Convention**                                         | **Example 1**                   | **Example 2**                             |
+|-------------------------|--------------------------------------------------------|---------------------------------|-------------------------------------------|
+| Nightly Source Builder  | {git-repo}-{git-branch}-source-release                 | foreman-develop-source-release  | hammer-cli-katello-master-source-release  |
+| Nightly Package Builder | {git-repo}-{git-branch}-package-release                | foreman-develop-package-release | hammer-cli-katello-master-package-release |
+| CI pipeline             | {repository}-{environment}-{optional-concern}-pipeline | foreman-nightly-rpm-pipeline    | foreman-nightly-deb-pipeline              |
+| Pull Request testing    | test_{git-repo}_{optional-concern}_pull_request        | katello-pr-test                 | foreman-packaging-rpm-pr-test             |
+
+## Job configurations
+
+### Testing develop
+
+All repos with an associated job that tests their master/develop branch should have a hook added to the repo to trigger immediate builds.
+
+To set up the hook, an org/repo admin must:
+
+* View the repository settings
+* Click *Webhooks*
+* Click *Add webhook*
+  * Payload URL: https://ci.theforeman.org/github-webhook/
+  * Content Type: application/json (default)
+  * Secret: add from secret store
+  * Just the push event
+
+### Pull request testing
+
+Core Foreman projects have GitHub pull requests tested on our Jenkins instance so it's identical to the way we test the primary development branches themselves.  Simpler and quieter projects (such as installer modules) should use Travis CI which supports PR testing and reduces the load on our own infrastructure.
```

Comment:
We no longer use Travis CI. Perhaps this should be dropped or at least changed to
```suggestion
Core Foreman projects have GitHub pull requests tested on our Jenkins instance so it's identical to the way we test the primary development branches themselves.  Simpler and quieter projects (such as installer modules) can use Github Actions instead. There is no obligation for Foreman projects to use Jenkins.
```